File: src/main/java/com/example/moexbondservice/service/utils/CorporateBondService.java
package com.example.moexbondservice.service.utils;

import com.example.moexbondservice.client.CorporateBondClient;
import com.example.moexbondservice.dto.BondDto;
import lombok.RequiredArgsConstructor;
import org.springframework.cache.annotation.Cacheable;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class CorporateBondService {
    private final BondParser bondParser;
    private final CorporateBondClient corporateBondClient;
    private final MoexBondValidator validator;

    @Cacheable(value = "corps")
    public List<BondDto> getCorporateBonds() {
        final String corporateRates = corporateBondClient.getBondsFromMoex();
        List<BondDto> bonds = bondParser.parse(corporateRates);
        validator.validateMoexLimitation(bonds);

        return bonds;
    }

}

File: src/main/java/com/example/moexbondservice/service/utils/CorporateBondService.kt
package com.example.moexbondservice.service.utils

import com.example.moexbondservice.client.CorporateBondClient
import com.example.moexbondservice.dto.BondDto
import org.springframework.cache.annotation.Cacheable
import org.springframework.stereotype.Service

@Service
open class CorporateBondService(private val bondParser: BondParser,
                                private val corporateBondClient: CorporateBondClient,
                                private val moexBondValidator: MoexBondValidator) {

    @Cacheable(value = ["corps"])
    open fun getCorporateBonds(): List<BondDto?>? {
        val corporateRates = corporateBondClient.getBondsFromMoex()
        val bonds = bondParser.parse(corporateRates)
        moexBondValidator.validateMoexLimitation(bonds)
        return bonds
    }
}
